src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.VectorSearch.cs

```diff
@@ -0,0 +1,445 @@
+        }
+
+        object HandleJsArray(JsArray jsArray)
+        {
+            var firstItem = jsArray[0];
+            if (firstItem.IsString())
+            {
+                var values = new object[jsArray.Length];
+                for (var i = 0; i < jsArray.Length; i++)
+                    values[i] = GenerateEmbeddings.FromBase64Array(vectorOptions, allocator, jsArray[i].AsString());
+
+                return values;
+            }
+
+            if (firstItem is JsArray)
+            {
+                var values = new object[jsArray.Length];
+                for (var i = 0; i < jsArray.Length; i++)
+                    values[i] = HandleJsArray(jsArray[i] as JsArray);
+
+                return values;
+            }
+
+            var len = (int)jsArray.Length;
+            var bufferSize = len * (vectorOptions.SourceEmbeddingType) switch
+            {
+                VectorEmbeddingType.Single => sizeof(float),
+                _ => sizeof(byte)
+            };
+
+            var memScope = allocator.Allocate(bufferSize, out Memory<byte> mem);
+            ref var floatRef = ref MemoryMarshal.GetReference(MemoryMarshal.Cast<byte, float>(mem.Span));
+            ref var sbyteRef = ref MemoryMarshal.GetReference(MemoryMarshal.Cast<byte, sbyte>(mem.Span));
+            ref var byteRef = ref MemoryMarshal.GetReference(mem.Span);
+
+            for (int i = 0; i < len; ++i)
+            {
+                var num = jsArray[i].AsNumber();
+                switch (vectorOptions.SourceEmbeddingType)
+                {
+                    case VectorEmbeddingType.Single:
+                        Unsafe.Add(ref floatRef, i) = (float)num;
+                        break;
+                    case VectorEmbeddingType.Int8:
+                        Unsafe.Add(ref sbyteRef, i) = Convert.ToSByte(num);
+                        break;
+                    default:
+                        Unsafe.AddByteOffset(ref byteRef, i) = Convert.ToByte(num);
+                        break;
+                }
+            }
+
+            return GenerateEmbeddings.FromArray(allocator, memScope, mem, vectorOptions, bufferSize);
+        }
+
+        object HandleStream(Stream stream)
+        {
+            var len = (int)stream.Length;
+            var memScope = allocator.Allocate((int)stream.Length, out Memory<byte> mem);
+            stream.ReadExactly(mem.Span);
+            return GenerateEmbeddings.FromArray(allocator, memScope, mem, vectorOptions, len);
+        }
+
+        bool IsBase64(object val) => CanTransformIntoString(val);
+
+        bool TryGetFirstNonNullElement(BlittableJsonReaderArray data, out object firstNonNull)
+        {
+            firstNonNull = data[0];
+
+            var i = 0;
+            while (IsNullValue(firstNonNull) && i < data.Length)
+                firstNonNull = data[i++];
+
+            return IsNullValue(firstNonNull) == false;
```

Comment:
Added support for cases where all elements are null and may contain different null representations (not just `null`).